src/gt4py/frontend/gtscript_frontend.py

```diff
@@ -964,22 +964,24 @@ def _eval_index(self, node: ast.Subscript) -> Optional[List[int]]:
         # Python 3.9 skips wrapping the ast.Tuple in an ast.Index
         tuple_or_constant = node.slice.value if isinstance(node.slice, ast.Index) else node.slice
 
-        # Python 3.8 still uses slice=ExtSlice
-        if isinstance(tuple_or_constant, ast.ExtSlice):
-            raise invalid_target
-
-        constant_nodes = gt_utils.listify(
-            tuple_or_constant.elts
-            if isinstance(tuple_or_constant, ast.Tuple)
-            else tuple_or_constant
+        tuple_or_expr = node.slice.value if isinstance(node.slice, ast.Index) else node.slice
+        index_nodes = gt_utils.listify(
+            tuple_or_expr.elts if isinstance(tuple_or_expr, ast.Tuple) else tuple_or_expr
         )
 
-        if any(isinstance(cn, ast.Slice) for cn in constant_nodes):
+        if any(isinstance(cn, ast.Slice) for cn in index_nodes):
             raise invalid_target
-        if any(isinstance(cn, ast.Ellipsis) for cn in constant_nodes):
+        if any(isinstance(cn, ast.Ellipsis) for cn in index_nodes):
             return None
         else:
-            return [ast.literal_eval(cn) for cn in constant_nodes]
+            index = []
+            for index_node in index_nodes:
+                try:
+                    offset = ast.literal_eval(index_node)
+                    index.append(offset)
+                except:
```

Comment:
Good idea, but it looks like quite a few different exceptions are raised by the standard (c.f. https://docs.python.org/3.10/library/ast.html#ast.literal_eval)

> Warning: It is possible to crash the Python interpreter with a sufficiently large/complex string due to stack depth limitations in Python’s AST compiler.

> It can raise ValueError, TypeError, SyntaxError, MemoryError and RecursionError depending on the malformed input.

Reply:
I see, in that case I would leave it as is and deal with this only if it turns out to skip errors too aggressively.